python/ray/train/v2/_internal/execution/worker_group/protocol.py

```diff
@@ -0,0 +1,32 @@
+from abc import abstractmethod
+from typing import Dict, Protocol
```

Comment:
What are the benefits of using `Protocol` here? Do we not directly inherit from this class anywhere?